go/vt/vterrors/code.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vterrors
+
+import (
+	"fmt"
+
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+)
+
+var (
+	VT03001 = errorWithState("VT03001", vtrpcpb.Code_INVALID_ARGUMENT, SyntaxError, "Aggregate functions take a single argument '%s'", "The planner accepts aggregate functions that take a single argument only.")
+	VT03002 = errorWithState("VT03002", vtrpcpb.Code_INVALID_ARGUMENT, ForbidSchemaChange, "Changing schema from '%s' to '%s' is not allowed", "This schema change is not allowed. You cannot change the keyspace of a table.")
+	VT03003 = errorWithState("VT03003", vtrpcpb.Code_INVALID_ARGUMENT, UnknownTable, "Unknown table '%s' in MULTI DELETE", "The specified table in this DELETE statement is unknown.")
+	VT03004 = errorWithState("VT03004", vtrpcpb.Code_INVALID_ARGUMENT, NonUpdateableTable, "The target table %s of the DELETE is not updatable", "You cannot delete something that is not a real MySQL table.")
+	VT03005 = errorWithState("VT03005", vtrpcpb.Code_INVALID_ARGUMENT, WrongGroupField, "Can't group on '%s'", "The planner does not allow grouping on such field.")
```

Comment:
> I prefer cannot and does not instead of can't and doesn't for readability.

This is resolved via 968c1d559e, for `VT03005`.

> What does it mean that "The planner does not allow grouping on such field." ??
What is it about the field that disallows grouping?

The short description of the error includes the AST nodes that leads to this error. I updated the long description to: `The planner does not allow grouping on certain field. For instance, aggregation function.`. @systay, let me know if this is accurate and valid. This is based on the behavior `checkForInvalidGroupingExpressions`: https://github.com/vitessio/vitess/blob/866506b338d34227d1ae5adbd30e731e304e26dc/go/vt/vtgate/planbuilder/operators/queryprojection.go#L578-L590